File: compiler/testData/loadJava/kotlinSignature/PropertyComplexTypes.java
package test;

import java.lang.String;
import java.util.ArrayList;

import jet.runtime.typeinfo.KotlinSignature;

public class PropertyComplexTypes<T> {
    @KotlinSignature("var genericType : T")
    public T genericType;

    @KotlinSignature("var listDefinedGeneric : ArrayList<String>")
    public ArrayList<String> listDefinedGeneric;

    @KotlinSignature("var listGeneric : ArrayList<T>")
    public ArrayList<T> listGeneric;

    @KotlinSignature("var listOfGenericList : ArrayList<ArrayList<T>>")
    public ArrayList<ArrayList<T>> listOfGenericList;
}

File: compiler/testData/loadJava/kotlinSignature/PropertyComplexTypes.kt
package test

import java.util.*

public open class PropertyComplexTypes<T>(p0 : T) : java.lang.Object() {
  public var genericType : T = p0
  public var listDefinedGeneric : ArrayList<String> = ArrayList<String>()
  public var listGeneric : ArrayList<T> = ArrayList<T>()
  public var listOfGenericList : ArrayList<ArrayList<T>> = ArrayList<ArrayList<T>>()
}
